File: src/main/java/org/hunter/TwoSum.java
package org.hunter;

import java.util.*;

public class TwoSum {

    public static void main(String [] args) {
        TwoSum twoSum = new TwoSum();
        System.out.println(twoSum.twoSum(new int []{5,6,7,19,4}, 11));
    }

    public List<List<Integer>> twoSum(int [] arr, int target) {
        List<List<Integer>> ans = new ArrayList<>();
        Arrays.sort(arr);
        int left = 0;
        int right = arr.length - 1;
        while(left < right) {
            int sum = arr[left] + arr[right];
            if(sum == target) {
                List<Integer> tmp = new ArrayList<>();
                tmp.add(arr[left]);
                tmp.add(arr[right]);
                ans.add(tmp);
                ++left;
                --right;
            }else if(sum > target) {
                --right;
            }else{
                ++left;
            }
        }
        return ans;
    }

}

File: js/twosum.js

const twosum = (nums, target) => {
    nums.sort()
    let left = 0
    let right = nums.length - 1
    while (left < right) {
        let sum = nums[left] + nums[right]
        if(sum > target) {
            --right
        }else if (sum < target) {
            ++left
        }else {
            return [nums[left], nums[right]]
        }
    }
    return []
}

console.log(twosum([1,2,3,4], 5))